packages/@lwc/babel-plugin-component/src/decorators/wire/transform.js

```diff
@@ -43,11 +43,22 @@ function getGeneratedConfig(t, wiredValue) {
     const configBlockBody = [];
     const configProps = [];
     const generateParameterConfigValue = memberExprPaths => {
+        // Note: When memberExprPaths ($foo.bar) has an invalid identifier (eg: foo..bar, foo.bar[3])
```

Comment:
I don't fully understand what this is.

Reply:
@caridy without this fix, the compiler will transform and input like this one: `@wire(FooAdapter, { bar: '$a.0b' })` into:
```js
   //..
  config: function ($cmp) {
     let v1 = $cmp.a;
     return {
        bar: v1 != null ? v1.0b : undefined
       };
  }
  //..
```
and `0b` is not a valid identifier, therefore it will result in an invalid syntax.

This fix, transforms it into:
```js
   //..
  config: function ($cmp) {
     let v1 = $cmp['a'];
     return {
        bar: v1 != null ? v1['0b'] : undefined
       };
  }
  //..
```
in order to preserve backward compatibility.

Ideally, this (and other similar cases) should result in a compilation error, but throwing at compile time for existing components (that weren't throwing) is not an option now.